lib/experimental.ts

```diff
@@ -1,15 +1,54 @@
-import { PluginResponse } from "./types";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+
+import { Docker } from "./docker";
+import * as staticModule from "./static";
+import { ImageType, PluginResponse } from "./types";
 
 export async function experimentalAnalysis(
-  options: any,
+  targetImage: string,
 ): Promise<PluginResponse> {
   // assume Distroless scanning
-  return distroless(options);
+  return distroless(targetImage);
 }
 
 // experimental flow expected to be merged with the static analysis when ready
-export async function distroless(options: any): Promise<PluginResponse> {
-  // assumption #1: the image is present in the local Docker daemon
+export async function distroless(targetImage: string): Promise<PluginResponse> {
+  const archiveDir = path.join(os.tmpdir(), "snyk-image-archives");
+  createTempDirIfMissing(archiveDir);
+  // TODO terrible way to convert slashes to anything else
+  // so we don't think it's a directory
+  const archiveFileName = `${targetImage.replace(/\//g, "__")}.tar`;
+  const archiveFullPath = path.join(archiveDir, archiveFileName);
+
   // assumption #2: the `docker` binary is available locally
-  throw new Error("not implemented");
+  const docker = new Docker(targetImage);
+  // assumption #1: the image is present in the local Docker daemon
+  await docker.save(targetImage, archiveFullPath);
+  try {
+    const scanningOptions = {
+      staticAnalysisOptions: {
+        imagePath: archiveFullPath,
+        imageType: ImageType.DockerArchive,
+        // TODO only for RPM, may be removed once we get rid of bdb dep
+        tmpDirPath: "",
+        distroless: true,
+      },
+    };
+
+    return staticModule.analyzeStatically(targetImage, scanningOptions);
+  } finally {
+    fs.unlinkSync(archiveFullPath);
+  }
+}
+
+function createTempDirIfMissing(archiveDir: string): void {
+  try {
+    fs.mkdirSync(archiveDir);
```

Comment:
why not checking if the folder exists instead of this?

Reply:
because this code is simpler.

Reply:
I'm really not in favour of throwing exception on a "normal" flow, exceptions should be used to errors :) 
what is not simple with checking with `fs.existsSync` before?

Reply:
you would still need to create it and check for errors, and you would probably want to also check for permissions.
throwing and catching exceptions is an okay pattern when it's controlled well. we're not writing real-time c++ code where a few milliseconds would be critical

Reply:
this specific bit was part of a snippet I found online to make sure a dir is created on older Node versions (on newer versions than the Plugin uses, you can pass a ```{ recursive: true }``` flag and just similarly to ```mkdir -p``` it would just work